src/zenml/zen_stores/sql_zen_store.py

```diff
@@ -11306,6 +11306,21 @@ def _attach_tags_to_resources(
 
         tag_schemas = []
         for tag in tags:
+            # Check if the tag is a string that can be converted to a UUID
+            if isinstance(tag, str):
+                try:
+                    tag_uuid = UUID(tag)
+                    tag_schema = self._get_schema_by_id(
```

Comment:
This should not be inside the try-catch, who knows if it raises a random `ValueError` that you would not want to ignore.

Reply:
Nice catch. I turned it into a `try-except-else` and the try clause only includes the UUID conversion.